contentgrid-appserver-domain/src/main/java/com/contentgrid/appserver/domain/DatamodelApiImpl.java

```diff
@@ -399,18 +402,24 @@ public void removeRelationItems(@NonNull Application application, @NonNull Relat
         );
     }
 
-    @RequiredArgsConstructor
     public static class ResponseOutputDataMapper {
-        private final List<Attribute> attributes;
+        private final Map<AttributeName, Attribute> attributes;
         private final AttributeMapper<Optional<AttributeData>, PlainDataEntry> attributeMapper;
 
+        public ResponseOutputDataMapper(List<Attribute> attributes,
+                AttributeMapper<Optional<AttributeData>, PlainDataEntry> attributeMapper) {
+            this.attributes = attributes.stream()
+                    .collect(Collectors.toMap( Attribute::getName, Function.identity()));
+            this.attributeMapper = attributeMapper;
+        }
+
         public InternalEntityInstance mapAttributes(@NonNull EntityData entityData) {
-            var data = LinkedHashMap.<String, PlainDataEntry>newLinkedHashMap(attributes.size());
-            for (var attribute : attributes) {
+            var data = LinkedHashMap.<String, PlainDataEntry>newLinkedHashMap(entityData.getAttributes().size());
+            for (var attribute : entityData.getAttributes()) {
```

Comment:
Can we perhaps rename `attribute` to `attributeData`? It is a bit confusing if everything is called attribute or attributes.
```suggestion
            for (var attributeData : entityData.getAttributes()) {
```